hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -519,9 +524,30 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingVersionConditionType)
 	}
 
+	// Check if machine payload needs to be updated (e.g. like when the pull secret has changed).
+	pullSecretBytes, err := r.getPullSecretBytes(ctx, hcluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	targetPayloadConfigHash := hashStruct(config + targetVersion + string(pullSecretBytes[:]))
+	isUpdatingPayloadConfigVersion := isUpdatingPayloadConfigVersion(nodePool, targetPayloadConfigHash)
+	if isUpdatingPayloadConfigVersion {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
```

Comment:
instead of creating a new annotation, can we compute the pull secret change as part of the existing targetConfigHash?

Reply:
Yeah no problem. I was being more granular, with respect to reporting status, in my line of thinking at the time.